scripts/start-utils

```diff
@@ -109,10 +117,107 @@ function log() {
   if isDebugging || isTrue "${LOG_TIMESTAMP:-false}"; then
     ts=" $(date --rfc-3339=seconds)"
   fi
-  echo "[init]${ts} $*"
+  echo -e "[init]${ts} $*"
   eval "$oldState"
 }
 
+function getSudoFromDistro(){
+  distro=$(getDistro)
+  command=
+  if [[ $distro == alpine ]]; then
+    command="su-exec"
+  else
+    command="gosu"
+  fi
+  echo $command
+}
+
+# Refer to https://unix.stackexchange.com/a/10065/102376
+function isTerminal() {
+  if test -t 1; then
+    # see if it supports colors...
+    ncolors=$(tput colors)
+    test -n "$ncolors" && test "$ncolors" -ge 8
+  else
+    return 1
+  fi
+}
+
+errorLogTag="[ERROR]"
+warningLogTag="[WARN]"
+
+if isTerminal; then
+  normal="$(tput sgr0)"
+  red="$(tput setaf 1)"
+  yellow="$(tput setaf 3)"
+  function getErrorColoredLogString() {
+    echo "${red}$errorLogTag $* ${normal}"
+  }
+  function getWarningColoredLogString() {
+    echo "${yellow}$warningLogTag $* ${normal}"
+  }
+else
+  function getErrorColoredLogString() {
+    echo "$errorLogTag $*"
+  }
+  function getWarningColoredLogString() {
+    echo "$warningLogTag $*"
+  }
+fi
+
+function error() {
+  echo -e "$(getErrorColoredLogString "$*")"
+}
+
+function logError() {
+  if isDebugging; then
+    set +x
+  fi
+  log "$(getErrorColoredLogString "$*")"
+  if isDebugging; then
+    set -x
+  fi
+}
+
+function warning() {
+  if isDebugging; then
+    set +x
+  fi
+  echo -e "$(getWarningColoredLogString "$*")"
+  if isDebugging; then
+    set -x
+  fi
+}
+
+function logWarning() {
+  log "$(getWarningColoredLogString "$*")"
+}
+
+function isNumeric() {
+  [[ $1 =~ ^[0-9]+$ ]]
+}
+
+function isNumericElseSetToDefault() {
+  local var_name="$1"
+  local default_value="$2"
+  
+  if ! isNumeric ${!var_name} ; then
+    eval "$var_name=$default_value"
+    export "$var_name"
```

Comment:
ShellCheck is reporting a warning here:

<img width="896" alt="image" src="https://github.com/user-attachments/assets/9776fb47-745b-44d0-8734-332c51979ed7">

Reply:
yes because i am doing the thing it thinks i do not want to do